core/src/main/java/org/jruby/runtime/callsite/CachingCallSite.java

```diff
@@ -249,7 +247,7 @@ public IRubyObject callIter(ThreadContext context, IRubyObject caller, IRubyObje
         }
     }
 
-    public CacheEntry retrieveCache(RubyClass selfType) {
+    public final CacheEntry retrieveCache(RubyClass selfType) {
         // This must be retrieved *once* to avoid racing with other threads.
```

Comment:
yes its mostly 'style' - those methods are not designed to get overriden on their own.
one most likely re-thing several of those marked final here to get it working properly ...